### ISSUE TITLE
Skip does not work on child list elements

### ISSUE THREAD

**Issue by johanwannheden** [modelmapper/modelmapper#360]
When mapping two objects of the same type, it appears that the `skip()` configuration is not honored for list element types.

Given
```
public class Foo {
  private String id;
  private List<Bar> bars;
  // public getters and setters provided by Lombok
}
```
with 
```
public class Bar {
  private String id;
  private String name;
  // public getters and setters provided by Lombok
}
```
and the following configuration
```
    ModelMapper mapper = new ModelMapper();

    mapper.createTypeMap(Foo.class, Foo.class).addMappings(new PropertyMap<Foo, Foo>() {
      @Override
      protected void configure() {
        skip().setId(null);
      }
    });
    mapper.createTypeMap(Bar.class, Bar.class).addMappings(new PropertyMap<Bar, Bar>() {
      @Override
      protected void configure() {
        skip().setId(null);
      }
    });
```
the test below fails:
```
    Foo foo1 = Foo.builder()
        .id("id-foo")
        .bars(ImmutableList.of(Bar.builder().id("id-bar").name("bar").build()))
        .build();
    Foo foo2 = Foo.builder()
        .id("id-boo")
        .bars(ImmutableList.of(Bar.builder().id("id-zar").name("car").build()))
        .build();

    mapper.map(foo1, foo2);

    assertEquals("id-foo", foo1.getId());
    assertEquals("id-boo", foo2.getId());

    assertEquals("bar", foo2.getBars().get(0).getName());
    assertEquals("id-zar", foo2.getBars().get(0).getId()); // java.lang.AssertionError: expected:<id-zar> but was:<null>
```

Is this a bug or a misconfiguration?

Thanks in advance for any hints.

Reproducer: https://github.com/johanwannheden/modelmapper-reproducer

**Comment by chhsiao90** [comment]
It's a bug and I had fixed this in #362

**Comment by chhsiao90** [comment]
Just release v2.1.0, please take a look.
